File: Easy/119. Pascal's Triangle II/119. Pascal's Triangle II.ts
function getRow(rowIndex: number): number[] {
    const row: number[] = new Array(rowIndex + 1).fill(0);
    row[0] = 1;

    for (let i = 1; i <= rowIndex; i++) {
        for (let j = i; j >= 1; j--) {
            row[j] += row[j - 1];
        }
    }

    return row;
}

File: Easy/119. Pascal's Triangle II/119. Pascal's Triangle II.rs
impl Solution {
    pub fn get_row(row_index: i32) -> Vec<i32> {
        let mut row = vec![1; row_index as usize + 1];

        for i in 0..(row_index as usize + 1) {
            let mut prev = 1;
            for j in 1..i {
                let temp = row[j];
                row[j] += prev;
                prev = temp;
            }
        }

        row
    }
}
